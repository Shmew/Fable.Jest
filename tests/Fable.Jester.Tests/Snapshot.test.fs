module SnapshotTests

open Fable.Jester
open Fable.ReactTestingLibrary
open Feliz

let testElement = React.functionComponent(fun () ->
    let (count, setCount) = React.useState(0)

    Html.div [
        Html.h1 count
        Html.button [
            prop.text "Increment"
            prop.onClick (fun _ -> setCount(count + 1))
        ]
    ])

let testElement2 = React.functionComponent(fun () ->
    let (count, setCount) = React.useState(0)

    Html.div [
        Html.h1 count
        Html.button [
            prop.text "Decrement"
            prop.onClick (fun _ -> setCount(count - 1))
        ]
    ])

Jest.describe("snapshot tests", (fun () ->
    Jest.test("using snap file", (fun () ->
        let render = RTL.render(testElement())

        Jest.expect(render.container.firstChild).toMatchSnapshot()
    ))

    Jest.test("using snap file", (fun () ->
        let render = RTL.render(testElement2())

        Jest.expect(render.container.firstChild).toThrowErrorMatchingSnapshot()
    ))
))
