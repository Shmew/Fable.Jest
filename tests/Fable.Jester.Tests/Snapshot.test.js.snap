// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests using snap file 1`] = `
<div>
  <h1>
    0
  </h1>
  <button>
    Increment
  </button>
</div>
`;
